botocore/credentials.py

```diff
@@ -550,19 +554,51 @@ def load(self):
         """
         if self._mapping['access_key'] in self.environ:
             logger.info('Found credentials in environment variables.')
-            access_key, secret_key = self._extract_creds_from_mapping(
-                self.environ, self._mapping['access_key'],
-                self._mapping['secret_key'])
-            token = self._get_session_token()
-            return Credentials(access_key, secret_key, token,
-                               method=self.METHOD)
+            fetcher = self._create_credentials_fetcher()
+            credentials = fetcher()
+            expiry_time = credentials['expiry_time']
+            if expiry_time is not None:
```

Comment:
I think it would good to have tests for these cases (fixed->refreshable, refreshable->fixed are not allowed).  Unlike other refreshable providers were it was unlikely to have missing keys across subsequent responses, this is the first refreshable provider that seems more prone to refresh errors.